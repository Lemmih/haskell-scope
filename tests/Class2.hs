module Class2 where

data String
data List a

class Show a where
  show :: a -> String
  showList :: List a -> String

  showList = show
