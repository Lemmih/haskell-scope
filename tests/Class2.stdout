Scope errors:

Definitions:
  Definition 1, from Class2.decl0:
    data [4mString[0m
  Definition 2, from Class2.decl1:
    data [4mList[0m a
  Definition 3, from Class2.decl1:
    data List [4ma[0m
  Definition 4, from Class2.decl2:
    class [4mShow[0m a where
  Definition 5, from Class2.decl2:
    class Show [4ma[0m where
  Definition 6, from Class2.decl2:
      [4mshow[0m :: a -> String
  Definition 7, from Class2.decl2:
      [4mshowList[0m :: List a -> String
  Definition 8, from Class2.decl2.match0:
      showList [4mlst[0m = show lst

Use sites:
  Definition used: 5
      show :: [4ma[0m -> String
  Definition used: 1
      show :: a -> [4mString[0m
  Definition used: 2
      showList :: [4mList[0m a -> String
  Definition used: 5
      showList :: List [4ma[0m -> String
  Definition used: 1
      showList :: List a -> [4mString[0m
  Definition used: 7
      [4mshowList lst = show lst[0m
  Definition used: 7
      [4mshowList[0m lst = show lst
  Definition used: 6
      showList lst = [4mshow[0m lst
  Definition used: 8
      showList lst = show [4mlst[0m

Exports:
  List, Data from Class2
  Show, Class from Class2
  String, Data from Class2
  show, Method from Class2
  showList, Method from Class2
